docs/src/reference-guide/components/graphql/search-filtering.md

```diff
@@ -0,0 +1,297 @@
+# Search and Filtering
+
+Currently, the Fuel indexer supports the following search and filtering operations:
+
+- ID selection
+- comparison
+- set membership
+- excluding null values
+
+Additionally, you can combine these operations using the `and` or `or` keywords, and invert operations by using the `not` keyword.
+
+To illustrate these operations, we'll use a `filter_example` entity that would typically return the following results with a regular, unfiltered query:
+
+```graphql
+query {
+  filter_example {
+    id
+    foo
+    possibly_null_bar
+    baz
+  }
+}
+```
+
+```json
+{
+  "data": [
+    {
+      "baz": 1,
+      "foo": "beep",
+      "id": 1,
+      "possibly_null_bar": 123
+    },
+    {
+      "baz": 5,
+      "foo": "boop",
+      "id": 2,
+      "possibly_null_bar": null
+    },
+    {
+      "baz": 1000,
+      "foo": "blorp",
+      "id": 3,
+      "possibly_null_bar": 456
+    }
+  ]
+}
+```
+
+> Note: It is important to note that you should practice sensible database design when filtering records. Apply database indicies to the underlying columns in order to make search operations more efficient; however, be advised that an overuse of database indicies will lead to degraded performance.
+
+## ID Selection
+
+You can query for a particular instance of an object by passing an ID value to with the `id` key. For example, let's say that you were storing blocks and you wanted details about a certain block, you would pass the requisite ID:
+
+```graphql
+query {
+  filter_example(id: 1) {
+    id
+    foo
+    possibly_null_bar
+    baz
+  }
+}
+```
+
+```json
+{
+  "data": [
+    {
+      "baz": 1,
+      "foo": "beep",
+      "id": 1,
+      "possibly_null_bar": 123
+    }
+  ]
+}
+```
+
+> Note: Remember that IDs currently must be of type `u64`, and as such, the ID selection operation will only allow for a `u64` value. We're open to changing this in the future.
+
+## Excluding Null Values
+
+You can store null values in your records if the corresponding entity fields are configured to allow for it. As such, you can exclude records that contain null values in a particular column or set of coulmns by using the `has` operator inside of a `filter` object.
```

Comment:
I don't mean to diminish your style! But I honestly feel this sentence reads better like:
"You can exclude records that ..."
Removing the "As such,"